File: aoc2024/java/src/test/java/com/peeekay/aoc2024/java/Day05Test.java
package com.peeekay.aoc2024.java;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class Day05Test {
    final Day05 day = new Day05(true);

    @Test
    void part1() {
        Assertions.assertEquals(143, day.part1());
    }

    @Test
    void part2() {
        Assertions.assertEquals(123, day.part2());
    }
}

File: test/com/peeekay/aoc2022/kotlin/Day05Test.kt
package com.peeekay.aoc2022.kotlin

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

@DisplayName("Day 05")
internal class Day05Test {

    @Test
    fun `Part1 with test data`() {
        val day = Day05(isTest = true)
        val answer = day.partOne()

        assertEquals("CMZ", answer)
    }

    @Test
    fun `Part2 with test data`() {
        val day = Day05(isTest = true)
        val answer = day.partTwo()

        assertEquals(4, answer)
    }
}